controller/controller.go

```diff
@@ -692,6 +692,7 @@ func (ctrl *ProvisionController) Run(_ <-chan struct{}) {
 			ctrl.leaderElectionNamespace,
 			strings.Replace(ctrl.provisionerName, "/", "-", -1),
 			ctrl.client.CoreV1(),
+			nil,
```

Comment:
It's not obvious why switching to glog depends on this change. It's probably related to some dependency change, but does that have to be in this PR?

Reply:
yes, master branch of this repo is following master branch k8s.io/*. so when k8s.io breaks something (i made this breaking change actually) we may need to update random things for builds to be green

Reply:
req'd for compilation on my end

Reply:
That would be an argument in favor of having `Gopk.toml` + `Gopkg.lock` file in the repo: with those present, someone who works on the source can focus on one aspect (like adding k8s.io/klog) without having to deal with unrelated changes.

Reply:
yes. I want to avoid doing that because in my experience imposing explicit version requirements for the library via a checked-in Gopkg.toml+lock can cause headaches for consumers. e.g. the Gopkg.toml will say this library needs k8s.io/* master branch but really it will work also with release-1.14...if another dependency needs release-1.14, can dep figure out to use release-1.14 or will it complain that there the needs are unsatisfiable? (bit of a contrived example)

Anyway, if master branch is supposed to track k8s.io/* master, ideally I should be periodically syncing it up, I'll look into adding at travis cron job let me know when something needs updating to reduce headaches for *contributors*

Reply:
You can have an empty `Gopk.toml` (or, better, a minimal one which just records the real constraints) and just record what's known to work in `Gopkg.lock`. Then people working with the source and Travis CI can use `Gopkg.lock` to fetch exactly the same source as the last person working on a dependency update (via `dep ensure` without `-update`). Downstream users will just ignore the `Gopkg.lock`.